src/XliffTasks/Model/ResxDocument.cs

```diff
@@ -72,5 +74,46 @@ public override void RewriteRelativePathsToAbsolute(string sourceFullPath)
                 }
             }
         }
+
+        /// <summary>
+        /// Attempts to match formatting placeholders as documented at
+        /// https://docs.microsoft.com/en-us/dotnet/standard/base-types/composite-formatting.
+        /// Explanation:
+        /// \{
+        ///    A placeholder starts with an open curly brace. Since curly braces are used in
+        ///    regex syntax we escape it to be clear that we mean a literal {.
+        ///
+        /// (\d+)
+        ///    The "index" component; one or more decimal digits. This is captured in a group
+        ///    to facilitate extracting the numeric value.
+        ///
+        /// (\,\-?\d+)?
+        ///    The optional "alignment" component. This is a comma, followed by an optional
+        ///    minus sign, followed by one or more decimal digits.
+        ///
+        /// (\:[^\}]+)?
+        ///    The optional "format string" componet. This is a colon, followed by one or more
+        ///    characters that aren't close curly braces.
+        ///
+        /// \}
+        ///    The close curly brace indicates the end of the placeholder.
+        /// </summary>
+        private static Regex s_placeHolderRegex = new Regex(@"\{(\d+)(\,\-?\d+)?(\:[^\}]+)?\}", RegexOptions.Compiled);
```

Comment:
Now I have two problems. Sorry, couldn't resist. ;)

Man, this is hard to read. Thank you for the docs. I'm looking into another approach, but not sure if it'll be better yet.

Reply:
I thought about writing my own parser, and decided that "regex + copious comments" was shorter and easier to follow.

Reply:
What about escaped `{{`?

(I would NOT recommend this, I found a HACK to use IFormatProvider + ICustomFormatter to count placeholders using string.Format: https://gist.github.com/nguerrera/dfd08d2ceaf38804cbcc5e60b1a48476 😆)

I'm looking at StringBuilder.AppendFormat now. Doesn't seem like the code would be very big if we stole it and replaced the appending with counting, but I'll give it a spin before endorsing that.

Reply:
[Our docs on Escaping Braces](https://docs.microsoft.com/en-us/dotnet/standard/base-types/composite-formatting#escaping-braces) basically say "don't do that" so I've chosen not to worry about "{{" and "}}".

Reply:
StringBuilder.AppendFormat hacking is messier than I'd hoped. :(